src/client/datascience/openNotebookBanner.ts

```diff
@@ -0,0 +1,46 @@
+import { inject, injectable, named } from 'inversify';
+import { Memento } from 'vscode';
+import { IExtensionSingleActivationService } from '../activation/types';
+import { IPythonExtensionChecker } from '../api/types';
+import { IApplicationEnvironment, IApplicationShell } from '../common/application/types';
+import { UseVSCodeNotebookEditorApi } from '../common/constants';
+import { GLOBAL_MEMENTO, IMemento } from '../common/types';
+import { DataScience } from '../common/utils/localize';
+import { isPythonNotebook } from './notebook/helpers/helpers';
+import { INotebookEditor, INotebookEditorProvider } from './types';
+
+const ShowedKernelMessageKey = 'ShowedKernelLocationMessage';
+
+@injectable()
+export class OpenNotebookBanner implements IExtensionSingleActivationService {
+    constructor(
+        @inject(INotebookEditorProvider) private notebookEditorProvider: INotebookEditorProvider,
+        @inject(IPythonExtensionChecker) private pythonExtensionChecker: IPythonExtensionChecker,
+        @inject(IApplicationEnvironment) private appEnv: IApplicationEnvironment,
+        @inject(IApplicationShell) private appShell: IApplicationShell,
+        @inject(UseVSCodeNotebookEditorApi) private useVSCodeNotebookEditor: boolean,
+        @inject(IMemento) @named(GLOBAL_MEMENTO) private mementoStorage: Memento
+    ) {}
+
+    public async activate() {
+        this.notebookEditorProvider.onDidOpenNotebookEditor(this.openedNotebook.bind(this));
+    }
+
+    private async openedNotebook(editor: INotebookEditor) {
+        if (
+            !this.pythonExtensionChecker.isPythonExtensionInstalled &&
+            editor.model.metadata &&
+            isPythonNotebook(editor.model.metadata)
+        ) {
+            await this.pythonExtensionChecker.showPythonExtensionInstallRecommendedPrompt();
+        } else if (
+            this.appEnv.channel === 'stable' &&
```

Comment:
Wouldn't insiders also probably want to see this? I'd imagine we're still getting new insiders users, or ones who didn't know about thiat.

Reply:
@claudiaregio you originally asked for stable only. Do you want to consider this for insiders too?